src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,70 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(String.format(
+                    "Can't save movie session: %s", movieSession), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movieSession;
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return Optional.ofNullable(session.get(MovieSession.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException(String.format(
+                    "Can't find movie session with id: %d", id), e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDateTime from,
+                                                    LocalDateTime to) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> findAvailableSessionsQuery = session.createQuery(
+                    "from MovieSession ms "
+                            + "left join fetch ms.movie m "
+                            + "where m.id= :id "
+                            + "and ms.showTime "
+                            + "between :from and :to", MovieSession.class);
+            findAvailableSessionsQuery.setParameter("id", movieId);
+            findAvailableSessionsQuery.setParameter("from", from);
+            findAvailableSessionsQuery.setParameter("to", to);
+            return findAvailableSessionsQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException(String.format(
+                    "Can't get available session for a movie by id: %d at this date: %s",
+                    movieId, from.toLocalDate()), e);
+        }
+    }
```

Comment:
The method 'findAvailableSessions' should not fetch all MovieSession entities and then filter them on the service layer. Instead, ensure that the query retrieves only the relevant MovieSession entities that fall within the specified day. You need to adjust the 'to' parameter to the end of the day of the 'from' parameter to limit the results to the correct day.